### PR TITLE
feat(rust, python): allow passing utc=True when parsing time-zone-naive date strings

### PR DESCRIPTION
**Pull request by MarcoGorelli**
On the way towards https://github.com/pola-rs/polars/issues/7198 (this should probably be done first)

On `master`, this errors:
```python
In [7]: pl.Series(['2020-01-01']).str.strptime(pl.Datetime, utc=True)
---------------------------------------------------------------------------
ComputeError: Cannot use 'utc=True' with tz-naive data. Parse the data as naive, and then use `.dt.convert_time_zone('UTC')`.
```

This PR allows for that:
```python
In [1]:  pl.Series(['2020-01-01']).str.strptime(pl.Datetime, utc=True)
Out[1]: 
shape: (1,)
Series: '' [datetime[μs, UTC]]
[
        2020-01-01 00:00:00 UTC
]
```

### CODE REVIEW

File: py-polars/tests/unit/datatypes/test_temporal.py [hunk 1]
```diff
@@ -1760,6 +1760,21 @@ def test_strptime_with_invalid_tz() -> None:
...
+            pl.Datetime("us", "foo"), "%Y-%m-%d %H:%M:%S", utc=True
+        )
+
+
+def test_strptime_unguessable_format() -> None:
+    with pytest.raises(
+        ComputeError,
+        match="Could not find an appropriate format to parse dates, please define a fmt",
+    ):
+        pl.Series(["foobar"]).str.strptime(pl.Datetime)
```
Comment (MarcoGorelli) [inline py-polars/tests/unit/datatypes/test_temporal.py:1777]: adding this as there wasn't any test for this to begin with

File: py-polars/tests/unit/namespaces/test_strptime.py [hunk 2]
```diff
@@ -93,14 +93,13 @@ def test_strptime_precision_with_time_unit(
...
-        ),
-    ):
-        pl.Series(["2020-01-01 00:00:00"]).str.strptime(pl.Datetime, fmt, utc=True)
+    result = (
+        pl.Series(["2020-01-01T00:00:00"])
+        .str.strptime(pl.Datetime, fmt, utc=True)
+        .item()
+    )
+    expected = datetime(2020, 1, 1, tzinfo=timezone.utc)
+    assert result == expected
```
Comment (MarcoGorelli) [inline py-polars/tests/unit/namespaces/test_strptime.py:102]: same example, but now it no longer errors

File: py-polars/polars/internals/construction.py [hunk 3]
```diff
@@ -51,7 +51,6 @@
     _PANDAS_AVAILABLE,
     _PYARROW_AVAILABLE,
     _check_for_numpy,
-    _check_for_pandas,
```
Comment (ritchie46) [inline py-polars/polars/internals/construction.py:54]: I don't think this was intentional.

**Review by ritchie46 — COMMENTED** [review]
Thanks! I have one remark.

File: py-polars/polars/internals/construction.py (hunk 3 above)
Comment (MarcoGorelli) [reply py-polars/polars/internals/construction.py:54]: I didn't do it manually, but `make pre-commit` removes it for me

Comment (MarcoGorelli) [reply py-polars/polars/internals/construction.py:54]: I think this should have been caught by https://github.com/pola-rs/polars/actions/runs/4270474083/jobs/7434291441, where the logs say

```
 Found 1 error (1 fixed, 0 remaining).
```
but it didn't fail CI

I'll take a look

Comment (ritchie46) [reply py-polars/polars/internals/construction.py:54]: Hmm... I also saw this change happening locally? Maybe we don't use that line?

Comment (MarcoGorelli) [reply py-polars/polars/internals/construction.py]: yup - mystery solved: https://github.com/pola-rs/polars/pull/7215/files
fetching and merging upstream/master fixes it here

**Pull request merged**